keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/PersistentEvent.kt

```diff
@@ -34,3 +34,24 @@ abstract class PersistentEvent {
     @JsonProperty("application") APPLICATION
   }
 }
+
+/**
+ * Common interface implemented by all [ResourceEvent]s and certain [ApplicationEvent]s that affect all the
+ * application's resources, such as pausing and resuming.
+ */
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  property = "type",
+  include = JsonTypeInfo.As.PROPERTY
+)
+@JsonSubTypes(
+  JsonSubTypes.Type(value = ResourceEvent::class),
+  JsonSubTypes.Type(value = ApplicationActuationPaused::class),
+  JsonSubTypes.Type(value = ApplicationActuationResumed::class)
+)
+interface ResourceHistoryEvent {
```

Comment:
The new `ResourceHistoryEvent` interface provides a common type for events of any scope that belong in the resource event history.